src/socket.c

```diff
@@ -894,16 +909,21 @@ CattaDnsPacket *catta_recv_dns_packet_ipv6(
             switch (cmsg->cmsg_type) {
 
                 case IPV6_HOPLIMIT:
-
-                    if (ret_ttl)
-                        *ret_ttl = (uint8_t) (*(int *) CMSG_DATA(cmsg));
-
+                    if (ret_ttl) {
+                      uint8_t cmsg_data_int8;
```

Comment:
I think there are some bugs here. The memcpy length exceeds 1 byte, but the destination is only 1 byte long.

Lets talk about this too...